src/unistd.rs

```diff
@@ -2524,6 +2527,51 @@ mod setres {
     }
 }
 
+#[cfg(any(target_os = "android", target_os = "linux"))]
+mod getres {
+    use crate::Result;
+    use crate::errno::Errno;
+    use super::{Uid, Gid};
+
+    /// Gets the real, effective, and saved uid.
+    ///
+    /// ([see getresuid(2)](http://man7.org/linux/man-pages/man2/getresuid.2.html))
+    ///
+    /// #Returns
+    ///
+    /// - `Ok((Uid, Uid, Uid))`: tuple of real, effective and saved uids on success.
+    /// - `Err(x)`: libc error code on failure.
+    ///
+    #[inline]
+    pub fn getresuid() -> Result<(Uid, Uid, Uid)> {
```

Comment:
Returning a tuple is a very old fashioned and failure-prone technique.  Better to return a structure.  That way the meaning is obvious even without consulting the documentation.  Something like:
```rust
struct ResUid {
    real: Uid,
    effective: Uid,
    saved: uid
}
```